core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2888,10 +2888,34 @@ class KafkaApis(val requestChannel: RequestChannel,
                 )
               }
             } else {
-              autoTopicCreationManager.createStreamsInternalTopics(topicsToCreate, requestContext);
+              // Compute group-specific expiration time for freshly cached errors
+              val sessionTimeoutMs = Option(groupConfigManager.groupConfig(streamsGroupHeartbeatRequest.data.groupId).orElse(null))
+                .map(_.streamsSessionTimeoutMs().toLong)
+                .getOrElse(config.groupCoordinatorConfig.streamsGroupSessionTimeoutMs().toLong)
+              val expirationTimeMs = time.milliseconds() + sessionTimeoutMs
```

Comment:
Two things about this

 - I believe we said that we want to use `heartbeatIntervalMs * 2` for the timeout, to make sure a client sees the topic creation errors.
 - I would pass an timeoutMs (a duration), not the expirationTimeMs (a time). We want to start the timeout from the point we receive the response, not here.